### ISSUE TITLE
Verbose Hover: Hover jumps when there is not enough space

### ISSUE THREAD

**Issue by benibenj** [microsoft/vscode#211063]
Testing #210904

When the hover grows and there is no more space at the top of the window the hover jumps to be displayed below the line for which the hover is provided. I'm not sure if there is a simple way to solve this problem, but the user experience would be much nicer if this would not happen.

https://github.com/microsoft/vscode/assets/44439583/281aa55d-bf45-4757-8a4f-1f07c908e665

**Comment by aiday-mar** [comment]
That is true. Before the jumping of the hover was not really an issue because there were no buttons you would repeatedly click, but now that you can repeatedly click the + or -, this impacts the user experience. 

Generally this would require some fundamental changes to how the editor hover is displayed. We would have to do either of the following:

- Either we should not allow the height to grow more than the maximum height above or below the cursor (depending on the hover placement setting). This is not ideal because a big hover could be compressed in a little space without a possibility to increase its height. 
- Or we would have to scroll the editor when the editor hover height increases in order to fit the hover vertically. This could be seen as undesirable behavior by users as they may not want the scroll top position to change as you'd hover on different symbols.

I'll mark this issue as a feature request and open for discussion.

**Comment by aiday-mar** [comment]
fix that was previously merged to fix this, caused a regression, so reopening this issue
